File: examples/python/client_example.py
#!/usr/bin/env python3

from phos.client import Client
import asyncio


async def main():
    client = Client()
    client.request('localhost', '1965', 'gemini://localhost/\r\n')

    await client.handshake()
    print('handshake done')

    (code, meta) = await client.header()
    print(f'code={code} meta={meta}')

    page = bytearray()
    while await client.recv_chunk():
        (chunk, size) = client.chunk()
        page[len(page):len(page)] = chunk[:size]
    print("body:")
    print(page.decode('utf-8'))

    await client.close()

if __name__ == '__main__':
    asyncio.run(main())

File: examples/lua/client_example.lua
local phos = require('phos')

local client = phos:new_client()
client:req('localhost', '1996', 'gemini://localhost/\r\n')

client:handshake_sync()
if client:response_sync() == -1 then
   error(client:err())
end

local code, meta = client:res();
print(string.format('code=%d meta=%s', code, meta))

while true do
   local buf, l = client:read_sync()
   if l == 0 then
      break
   elseif l == -1 then
      error(client:err())
   else
      print(buf)
   end
end

client:close_sync()
